### ISSUE TITLE
Error saving Fallout 3 mod as a Fallout New Vegas mod after sorting Masters

### ISSUE THREAD

**Issue by HersheyWaffle** [TES5Edit/TES5Edit#1206]
**What is the version of xEdit you are using?**
FNVEdit 4.0.4 (5B1C2C51)

**Describe the bug**
It seems like after exceeding a certain number of records, a plugin refuses to save after I add TTW and its masters then sort masters. Adding masters is fine and saves normally, but sorting the masters produces this error:

"[00:11] Background Loader: finished
[00:00] Start: sorting masters
[00:01] Done: sorting masters, Elapsed Time: 00:01
[00:00] Saving: Evergreen.esm.save.2023_09_30_23_27_33
[00:00] Error saving Evergreen.esm.save.2023_09_30_23_27_33: Assertion failure (P:\Projects\xEdit-4.0.x\wbImplementation.pas, line 14861)
[00:00] Errors have occured. At least one file was not saved."

The error seems to happen to plugins with over 2000 records. I tested with a plugin with 8000, 20000, and 2400 records, and all of them failed to save, producing the same error message. However, by reducing that last one's record count to 1712 by haphazardly removing some dialogue, script and npc records, it managed to save without error.

**To reproduce**
1. Load TaleOfTwoWastelands.esm (and its Masters) and a Fallout 3 plugin with over 2000 records
2. Click on 'Add Masters'
3. Select all
4. Click on 'Sort Masters (To match current load order)'
5. Save or exit
6. Scroll down to messages
7. See error

**Expected behavior**
To save without issue after sorting.

**Screenshots**
N/A

**Additional context**
N/A

**Comment by HersheyWaffle** [comment]
Never mind, it was because of HITMEs caused by dialogue edits.

**Comment by Superfly-Johnson** [comment]
No, this bug is absolutely real and doesn't happen with FNVEdit 4.0.3, only 4.0.4 and above cause this particular error.

I would suggest reopening the issue

In particular it's the dialogue subrecords that are present in masters that are problematic, in particular the greeting and radiochatter dialogues. I don't know why it's causing it, in 4.0.4 I needed to move those records to a dummy dialogue container and then change the formid after adding the masters, in 4.0.3 it simply worked.

**Comment by ElminsterAU** [comment]
Please provide a reproducable testcase (all required files that are not vanilla FNV provided, and exact reproduction steps) which shows this error in the latest version of xEdit available currently in the #xedit-builds channel on the xEdit discord.

**Comment by Superfly-Johnson** [comment]
> Please provide a reproducable testcase (all required files that are not vanilla FNV provided, and exact reproduction steps) which shows this error in the latest version of xEdit available currently in the #xedit-builds channel on the xEdit discord.

Alright, I was working on porting Vault 101 revisited to TTW (still am for the record).

So I downloaded the latest plugin (version 1.0.5) https://www.nexusmods.com/fallout3/mods/13308
Loaded it in xEdit 4.0.4c alongside TaleofTwoWastelands.esm and all of its masters
Pressed the Add masters dialogue, it saves then.
Pressed the sort masters dialogue, it refuses to save because certain dialogue subrecords are now HITMEs.

The same issue doesn't happen in 4.0.3, where it simply works as intended.

I tried the same process in 4.1.4p and the dialogue to add masters was simply broken in that version for new vegas, no idea why. I hope this helps.

EDIT: Adding the masters works after cleaning them, somehow. It also saves properly on 4.1.4p
